### PR TITLE
Add DataflowStartYamlJobOperator

### PR DESCRIPTION
**Pull request by moiseenkov**
Add DataflowStartYamlJobOperator as a replacement option for DataflowStartSqlJobOperator.

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/dataflow.py [hunk 1]
```diff
@@ -938,6 +937,110 @@ def launch_job_with_flex_template(
...
+            gcp_flags.update({"impersonate-service-account": impersonation_account})
+
+        return [
+            "gcloud",
+            "dataflow",
+            "yaml",
+            "run",
+            job_name,
+            *(beam_options_to_args(gcp_flags)),
+        ]
```
Comment (shahar1) [inline airflow/providers/google/cloud/hooks/dataflow.py]: This logic is taken from `DataflowHook.start_sql_job`. Maybe you could extract it to a utility function that will be used by both? (with the required adjustments)

File: airflow/providers/google/cloud/operators/dataflow.py [hunk 2]
```diff
@@ -945,6 +948,11 @@ def on_kill(self) -> None:
             )
 
 
+@deprecated(
+    reason="DataflowStartSqlJobOperator is deprecated and will be removed after 31.01.2025. "
+    "Please use DataflowStartYamlJobOperator instead.",
+    category=AirflowProviderDeprecationWarning,
+)
```
Comment (shahar1) [inline airflow/providers/google/cloud/operators/dataflow.py]: Please check that it complies with the format that you suggested in https://github.com/apache/airflow/pull/41637 :)

Not a blocker for merging the current PR, just to reduce future work.

**Review by shahar1 — COMMENTED** [review]
Looks good overall!
I got a small comment regarding reusability.

File: airflow/providers/google/cloud/hooks/dataflow.py (hunk 1 above)
Comment (moiseenkov) [reply airflow/providers/google/cloud/hooks/dataflow.py]: Thanks for the suggestion. Please take a look at the update.

File: airflow/providers/google/cloud/operators/dataflow.py (hunk 2 above)
Comment (moiseenkov) [reply airflow/providers/google/cloud/operators/dataflow.py]: Good point, thank you! The deprecation is updated.

**Comment by moiseenkov** [comment]
> Looks good overall!
> I got a small comment regarding reusability.

@shahar1 , 
Thanks for the review. Please take a look at the fixes.

**Review by potiuk — APPROVED** [review]

**Review by shahar1 — APPROVED** [review]

**Pull request merged**